packages/headless/src/controllers/facet-manager/headless-facet-manager.ts

```diff
@@ -21,12 +22,7 @@ export function buildFacetManager(engine: Engine<SearchSection>) {
      * @returns FacetManagerPayload[].
      */
     sort<T>(facets: FacetManagerPayload<T>[]) {
-      const payloadMap: Record<string, FacetManagerPayload<T>> = {};
-      facets.forEach((f) => (payloadMap[f.facetId] = f));
-
-      return this.state.facetIds
-        .map((id) => payloadMap[id])
-        .filter((payload) => payload !== undefined);
```

Comment:
I extracted this to a new util function that is also used in `search-actions.ts`